rs/consensus/src/consensus/metrics.rs

```diff
@@ -385,29 +385,53 @@ impl FinalizerMetrics {
     }
 }
 
-pub struct NotaryMetrics {
-    pub time_to_notary_sign: HistogramVec,
+pub(crate) struct NotaryMetrics {
+    time_to_notary_sign: HistogramVec,
+    adjusted_notary_delay: HistogramVec,
 }
 
 impl NotaryMetrics {
-    pub fn new(metrics_registry: MetricsRegistry) -> Self {
+    pub(crate) fn new(metrics_registry: &MetricsRegistry) -> Self {
         Self {
             time_to_notary_sign: metrics_registry.histogram_vec(
                 "consensus_time_to_notary_sign",
-                "The duration since round start for replicas to notary-sign a block, labelled by ranks",
-                vec![0.0, 0.2, 0.4, 0.6, 0.8, 1.0, 1.2, 1.4, 1.6, 1.8, 2.0, 2.2, 2.4, 2.6, 2.8, 3.0, 3.5, 4.0, 4.5, 5.0, 6.0, 8.0, 10.0, 15.0, 20.0],
+                "The duration since round start for replicas to notary-sign a block, \
+                labelled by ranks",
+                vec![
+                    0.0, 0.2, 0.4, 0.6, 0.8, 1.0, 1.2, 1.4, 1.6, 1.8, 2.0, 2.2, 2.4, 2.6, 2.8, 3.0,
+                    3.5, 4.0, 4.5, 5.0, 6.0, 8.0, 10.0, 15.0, 20.0,
+                ],
+                &["rank"],
+            ),
+            adjusted_notary_delay: metrics_registry.histogram_vec(
```

Comment:
Is there a specific thing you want to debug? I feel like for rank 0 blocks, most of the time, this is just going to return the initial notary delay